ui/app/AppLayouts/Wallet/panels/DAppsWorkflow.qml

```diff
@@ -206,7 +207,10 @@ DappsComboBox {
                 }
             }
 
-            onClosed: Qt.callLater( () => sessionRequestLoader.active = false)
+            onClosed: {
```

Comment:
When the dialog is `accepted` or `rejected` it will first signal the close and then the accepted/rejected (Looks like a Qt/StatusQ bug to me..). Rejecting by default on close means to reject all the requests. Even when accept was emitted.

I think this will do it:

```
            onClosed: Qt.callLater( () => {
                   if (result !== Dialog.Accepted) rejectRequest()
                  sessionRequestLoader.active = false
            })
```

Reply:
Indeed good catch. I forgot about the result property. I will try to use it also instead of the `requestRejected` flag to avoid a double disconnection. Thanks for the heads up.

Reply:
> `if (result !== Dialog.Accepted) rejectRequest()`

This doesn't work. The implementation has to use done() instead of explicit functions for the result to be set; see https://doc.qt.io/qt-5/qml-qtquick-controls2-dialog.html#done-method

However, i fixed this tricky case by postponing the call.